File: 프로그래머스/LV2/기능개발.py
def solution(progresses, speeds):
    answer = []
    day = []
    max_num = 0
    for i in range(len(progresses)):
        if (100 - progresses[i]) % speeds[i] == 0:
            day.append((100-progresses[i]) // speeds[i])
        else:
            day.append((100 - progresses[i]) // speeds[i] + 1)
    for d in day:
        if max_num < d:
            max_num = d
            answer.append(1)
        else:
            answer[-1] += 1
    return answer

File: 프로그래머스/LV2/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] dayOfend = new int[100];
        int day = -1;
        for(int i=0; i<progresses.length; i++) {
            while(progresses[i] + (day*speeds[i]) < 100) {
                day++;
            }
            dayOfend[day]++;
        }
        for (int i=0; i<dayOfend.length; i++) {
            System.out.println(dayOfend[i]);
        }
        return Arrays.stream(dayOfend).filter(i -> i!=0).toArray();
    }
}

